src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemResponseBuilder.cs

```diff
@@ -124,6 +124,26 @@ public RetrieveWorkitemResponse BuildRetrieveWorkitemResponse()
         return new RetrieveWorkitemResponse(status, _dataset, _message);
     }
 
+    /// <inheritdoc />
+    public UpdateWorkitemResponse BuildUpdateWorkitemResponse(string workitemInstanceUid = null)
+    {
+        Uri url = null;
+        WorkitemResponseStatus status = WorkitemResponseStatus.Failure;
+
+        if (!_dataset.TryGetSingleValue<ushort>(DicomTag.FailureReason, out var failureReason)
+            && !string.IsNullOrWhiteSpace(workitemInstanceUid))
+        {
+            status = WorkitemResponseStatus.Success;
+            url = _urlResolver.ResolveRetrieveWorkitemUri(workitemInstanceUid);
+        }
+        else if (failureReason == FailureReasonCodes.UpsUpdateConflict)
+        {
+            status = WorkitemResponseStatus.Conflict;
+        }
+
+        return new UpdateWorkitemResponse(status, url, _message);
```

Comment:
somewhere we are setting a success message that gets passed as a Warning header in the response.